bodo/io/stream_parquet_write.py

```diff
@@ -539,6 +531,10 @@ class ParquetWriterAppendInfer(AbstractTemplate):
     def generic(self, args, kws):
         pysig = numba.core.utils.pysignature(parquet_writer_append_table)
         folded_args = bodo.utils.transform.fold_argument_types(pysig, args, kws)
+        # Update state type in signature to include build table type from input
+        input_table_type = folded_args[1]
+        new_state_type = ParquetWriterType(input_table_type)
+        folded_args = (new_state_type, *folded_args[1:])
```

Comment:
This pattern also seems to be duplicated?  Should be its own function in StreamingStateType?

Reply:
Same as above.